salinity_restoring_generation/make_salt_sfc_restore_from_regridded_woa.py

```diff
@@ -0,0 +1,174 @@
+
+The script creates an output file 'salt_sfc_restore.nc' which contains the smoothed 
+and concatenated salinity data for 12 months.
+
+Usage:
+    python make_salt_sfc_restore_from_regridded_woa.py <input_directory> <output_directory>
+
+Example:
+    python make_salt_sfc_restore_from_regridded_woa.py /path/to/input/dir /path/to/output/dir
+
+Command-line arguments:
+    - input_directory: The directory containing the initial conditions NetCDF files.
+    - output_directory: The directory where the output smoothed and concatenated NetCDF file will be saved.
+"""
+
+import xarray as xr
+import numpy as np
+from scipy.ndimage import uniform_filter
+import argparse
+import netCDF4 as nc
+from pathlib import Path
+import sys
+
+# Add the root path for the common scripts
+path_root = Path(__file__).parents[1]
+sys.path.append(str(path_root))
+
+from scripts_common import get_provenance_metadata, md5sum
+
+
+def smooth2d(src):
+    tmp_src = np.ndarray((src.shape[0] + 6, src.shape[1]))
+
+    # Window size
+    ws = 3
+
+    tmp_src[ws:-ws, :] = src[:, :]
+    tmp_src[:ws, :] = src[-ws:, :]
+    tmp_src[-ws:, :] = src[:3, :]
+
+    dest = uniform_filter(tmp_src, size=ws, mode="nearest")
+    return dest[ws:-ws, :]
+
+
+def main(input_path, output_path):
+    # Variable to be processed
+    variable_to_smooth = "salt"
+    variable_to_exclude = "temp"
+
+    # Template for file paths
+    file_template = f"{input_path}/woa23_ts_{{:02d}}_mom025.nc"
+
+    # Initialize lists to store data
+    monthly_data = []
+    time_values = []
+
+    # Loop over each month and open the corresponding file
+    for month in range(1, 13):
+        # Construct the file name
+        file_path = file_template.format(month)
+
+        try:
+            # Open the dataset
+            ds = xr.open_dataset(file_path)
+
+            # Process the variable to be smoothed
+            if variable_to_smooth in ds.data_vars:
+                data = ds[variable_to_smooth].isel(ZT=0).load()
+                data_np = np.squeeze(data.values)
+                # Append data to list
+                monthly_data.append(data_np)
+                time_values.append(
+                    ds["time"].values[0]
+                )  # Extract the actual time value
+        except FileNotFoundError:
+            print(f"File not found: {file_path}")
+
+    # Concatenate the data across all months along the time dimension
+    concatenated_data = np.stack(monthly_data, axis=0)
+
+    # Loop through each month for smoothing
+    smoothed_monthly_data = []
+    for i in range(concatenated_data.shape[0]):
+        # Smooth each month's data
+        data_np_smoothed = smooth2d(concatenated_data[i, :, :])
+        smoothed_monthly_data.append(data_np_smoothed)
+
+    smoothed_data_np = np.stack(smoothed_monthly_data, axis=0)
+
+    # Create a new NetCDF file using netCDF4
+    output_file = f"{output_path}/salt_sfc_restore.nc"
+    with nc.Dataset(output_file, "w", format="NETCDF4") as ncfile:
```

Comment:
If you use xarray suggestions above, the only extra attributes you need are these (i think):

        salt_var.standard_name = "sea_water_salinity"
        salt_var.long_name = "Smoothed sea water salinity at level 0m"
        salt_var.units = "1"

you should just be able to set those and then say

salt_smoothed_da.to_netcdf()